extension/js/common/platform/store/contact-store.ts

```diff
@@ -295,13 +295,17 @@ export class ContactStore extends AbstractStore {
     }
   }
 
+  private static getPubkeyId = (pubkey: Key): string => {
+    return (pubkey.type === 'x509') ? (pubkey.id + '-X509') : pubkey.id;
+  }
```

Comment:
Ah - you are suffixing it only for storage, not for the actual `fingerprint` in key. As long as this doesn't leak out of the ContactStore (this method stays private) then this looks good.

Reply:
Exactly